airflow/providers/databricks/hooks/databricks_sql.py

```diff
@@ -243,11 +244,15 @@ def run(
 
     @staticmethod
     def _make_serializable(result):
-        """Transform the databricks Row objects into JSON-serializable lists."""
+        """Transform the databricks Row objects into JSON-serializable namedtuple."""
+        columns: list[str] | None = None
         if isinstance(result, list):
-            return [list(row) for row in result]
+            columns = result[0].__fields__
+            row_object = namedtuple("Row", columns)
```

Comment:
The same testcase from earlier PR would still fail here, right? 

```
statement = f"DESCRIBE TABLE {table.name};"
hook.run(statement, parameters=parameters, handler=lambda x: x.fetchall())
```
with this code hook.run() will return [[val1, val2, val3],[val1, val2, val3], [val1, val2, val3]] instead of [row(), row(), row()]

Reply:
Agree with @utkarsharma2  comment, we still need to update the code in this PR.

Reply:
yep

Reply:
Indeed, this PR is not about preventing the breaking change. But a proposal to break less workflows - _the simple tuple/list is too agressive_ - and reach the goal of [this ADR](https://github.com/apache/airflow/blob/main/airflow/providers/common/sql/doc/adr/0002-return-common-data-structure-from-dbapihook-derived-hooks.md).

The databricks.sql.Row object is a subclass of tuple, which implements:
- a dict-like interface -> breaking change - not available in a namedtuple
- a 'asDict()' method -> breaking change - but namedtuple implements ['_asdict()'](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._asdict)
- fields accessible by name -> supported by namedtuple

Thus users of the dict interface have to rewrite their code. Is it an issue with a major bump version?

Reply:
Yes. We should avoid breaking change like that

Reply:
I put the serializable namedtuple behind a `return_serializable` flag, which is False by default. With a warning for a future release of the provider, where it would turn to True by default. And eventually later the flag can disappear too to only return the namedtuple.

What do you think ?